pkg/bisect/generic/array.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2023 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package generic
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type ArrayConfig[T any] struct {
+	Pred      func([]T) (bool, error)
+	PredLimit int
+	MaxChunks int
+	Logf      func(string, ...interface{})
+}
+
+// Array performs a generic binary-search based minimization algorithm.
+// The aim is to remove as many invidiual elements as possible while still
+// ensuring that Pred(concatenated remaining parts) == true.
+func Array[T any](config ArrayConfig[T], startElements ...[]T) ([]T, error) {
```

Comment:
Originally it was in `pkg/bisect` and calling `bisect.Array()` was in line with Go style. But then I had to move it to another package to avoid circular dependencies..

Renamed to `MinimizeSlice`.